internal/ingress/annotations/relativeredirects/main.go

```
@@ -0,0 +1,76 @@
+limitations under the License.
+*/
+
+package relativeredirects
+
+import (
+	networking "k8s.io/api/networking/v1"
+
+	goErrors "errors"
+
+	"k8s.io/ingress-nginx/internal/ingress/annotations/parser"
+	"k8s.io/ingress-nginx/internal/ingress/errors"
+	"k8s.io/ingress-nginx/internal/ingress/resolver"
+)
+
+const (
+	relativeRedirectsAnnotation = "relative-redirects"
+)
+
+var relativeRedirectsAnnotations = parser.Annotation{
+	Group: "redirect",
+	Annotations: parser.AnnotationFields{
+		relativeRedirectsAnnotation: {
+			Validator:     parser.ValidateBool,
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `If enabled, redirects issued by nginx will be relative. See https://nginx.org/en/docs/http/ngx_http_core_module.html#absolute_redirect`,
+		},
+	},
+}
+
+type relativeRedirect struct {
+	r                resolver.Resolver
+	annotationConfig parser.Annotation
+}
+
+func (ar relativeRedirect) GetDocumentation() parser.AnnotationFields {
+	return ar.annotationConfig.Annotations
+}
+
+func (ar relativeRedirect) Validate(anns map[string]string) error {
+	maxrisk := parser.StringRiskToRisk(ar.r.GetSecurityConfiguration().AnnotationsRiskLevel)
+	return parser.CheckAnnotationRisk(anns, maxrisk, relativeRedirectsAnnotations.Annotations)
+}
+
+// NewParser creates a new relativeRedirects annotation parser
+func NewParser(r resolver.Resolver) parser.IngressAnnotation {
+	return relativeRedirect{
+		r:                r,
+		annotationConfig: relativeRedirectsAnnotations,
+	}
+}
+
+func (ar relativeRedirect) Parse(ing *networking.Ingress) (interface{}, error) {
+	val, err := parser.GetBoolAnnotation(relativeRedirectsAnnotation, ing, ar.annotationConfig.Annotations)
+
+	// A missing annotation is not a problem, just use the default which is `absolute_redirects on`
+	if goErrors.Is(err, errors.ErrMissingAnnotations) {
+		return false, nil // default is false
```

why not return the error?

---

fixed